From 748f932ca30a75afc4d179296bb56b3cad5c6fec Mon Sep 17 00:00:00 2001
From: Djamil Elaidi <d-elaidi@ti.com>
Date: Tue, 8 Mar 2011 18:27:27 +0100
Subject: OMAP2+: hwmod: Add smart-standby wakeup support

The commit c138b539f02170a6cb399867d930a1568c365f91 was adding
the wakeup support for new OMAP4 IPs. This support is incomplete for
busmaster IPs that need as well to use smart-standby with wakeup.

This new standbymode is suported on HSI and USB_HOST_FS for the moment.

Add the new MSTANDBY_SMART_WKUP flag to mark the IPs that support this
capability.

Enable this new mode when applicable in _enable_wakeup, _disable_wakeup,
_sysc_enable and _sysc_idle.

The omap_hwmod_44xx_data.c will have to be updated to add this new flag.

Change-Id: I14935dcd0544a54f7684120932aa6e8b462d97d3
Signed-off-by: Benoit Cousson <b-cousson@ti.com>
Signed-off-by: Djamil Elaidi <d-elaidi@ti.com>
Reported-by: Djamil Elaidi <d-elaidi@ti.com>
Acked-by: Benoit Cousson <b-cousson@ti.com>
Cc: Paul Walmsley <paul@pwsan.com>
Cc: Kevin Hilman <khilman@deeprootsystems.com>
---
 arch/arm/mach-omap2/omap_hwmod.c             |   36 ++++++++++++++++++++------
 arch/arm/plat-omap/include/plat/omap_hwmod.h |    2 +-
 2 files changed, 29 insertions(+), 9 deletions(-)

Index: kernel/arch/arm/mach-omap2/omap_hwmod.c
===================================================================
--- kernel.orig/arch/arm/mach-omap2/omap_hwmod.c	2011-04-01 18:30:35.000000000 +0900
+++ kernel/arch/arm/mach-omap2/omap_hwmod.c	2011-04-02 18:10:05.000000000 +0900
@@ -781,8 +781,6 @@
 		_set_module_autoidle(oh, a_idlemode, &v);
 		_write_sysconfig(v, oh);
 	}
-
-
 }
 
 /**
Index: kernel/arch/arm/plat-omap/include/plat/omap_hwmod.h
===================================================================
--- kernel.orig/arch/arm/plat-omap/include/plat/omap_hwmod.h	2011-04-01 18:30:35.000000000 +0900
+++ kernel/arch/arm/plat-omap/include/plat/omap_hwmod.h	2011-04-02 18:10:05.000000000 +0900
@@ -79,7 +79,6 @@
 #define HWMOD_IDLEMODE_FORCE		(1 << 0)
 #define HWMOD_IDLEMODE_NO		(1 << 1)
 #define HWMOD_IDLEMODE_SMART		(1 << 2)
-/* Slave idle mode flag only */
 #define HWMOD_IDLEMODE_SMART_WKUP	(1 << 3)
 
 /**
